engine/src/test/java/io/camunda/zeebe/engine/state/deployment/FormStateTest.java

```diff
@@ -96,6 +96,68 @@ void shouldReturnEmptyIfNoFormIsDeployedForFormKey() {
     assertThat(persistedForm).isEmpty();
   }
 
+  @Test
+  void shouldNotFindFormAfterDeletion() {
+    // given
+    final var form = sampleFormRecord();
+    formState.storeFormRecord(form);
+
+    assertThat(formState.findLatestFormById(form.getFormIdBuffer(), form.getTenantId()))
+        .isNotEmpty();
+
+    // when
+    formState.deleteForm(form);
+
+    // then
+    assertThat(formState.findLatestFormById(form.getFormIdBuffer(), form.getTenantId())).isEmpty();
+  }
+
+  @Test
+  void shouldNotFindVersion2AsLatestFormAfterDeletion() {
+    // given
+    final var formV1 = sampleFormRecord();
+    final var formV2 = sampleFormRecord(2, formV1.getFormKey());
```

Comment:
❌ You are giving `formV2` the same key as `formV1`. This is not a realistic case as a newly deployed form will always have a new key as well. Only the `formId` should be the same.

This doesn't impact the test as it is currently. But if you would try to find `formV1` by key you would not get any result.

Reply:
Same for the tests below this one.

Reply:
You are right, but I don't care about the `formKey` in this tests. But I will update the form key in order to represent a realistic case 👍